New app-action: delay

mikesealey opened the pull request:
## Description
Adds new app-action - Delay. This will help with an issue for the "Scroll to field" action when used in on-screen-load actions, which was likely caused by the form not being loaded by the time the on-screen-load actions have started running.

## Addresses
- https://github.com/Budibase/budibase/issues/16378
- https://github.com/Budibase/budibase/issues/16379

## App Export
[Delay-export-1751646728552.tar.gz](https://github.com/user-attachments/files/21061997/Delay-export-1751646728552.tar.gz)

## Screenshots
![image](https://github.com/user-attachments/assets/de52333a-802f-4999-bfca-de1d10b9eedc)

https://jam.dev/c/8b90af5d-979f-4d46-b3f2-e34e41babeea


## Launchcontrol

Adds new app-action: Delay

mikesealey committed:
initial commit

mikesealey committed:
passes value to function

mikesealey committed:
styling

aptkingston commented on packages/client/src/utils/buttonActions.js:
```diff
@@ -514,6 +514,10 @@ const rowActionHandler = async action => {
 }
 
+const delayHandler = async action => {
+  return new Promise(resolve => setTimeout(resolve, action.parameters.duration))
```
This will work regardless of what value `duration` is, but we should probably sanity check it anyway. There's also no guarantee that it's a number - so I would use a normal number input for the setting instead of a drawer bindable input, and also do `action.parameters.duration || 0` here to account for null/undefined/empty string.

mikesealey committed:
moves scroll to field function to buttonActions

mikesealey committed:
removes unused function from innerForm

mikesealey committed:
Merge branch 'master' into budi-9411-app-action-delay

mikesealey committed:
reinstates incorrectly removed function

mikesealey committed:
removes logs

mikesealey committed:
establishes default value

mikesealey committed:
better placeholder

mikesealey committed:
Merge branch 'master' into budi-9411-app-action-delay

mikesealey committed:
Merge branch 'master' into budi-9411-app-action-delay

mikesealey committed:
consolidates scrollToField functions outside forms

mikesealey committed:
Merge branch 'master' of https://github.com/budibase/budibase into budi-9411-app-action-delay

mikesealey committed:
Merge branch 'budi-9411-app-action-delay' of https://github.com/budibase/budibase into budi-9411-app-action-delay

mikesealey committed:
Merge branch 'master' into budi-9411-app-action-delay

deanhannigan committed:
Merge branch 'master' into budi-9411-app-action-delay

deanhannigan reviewed (changes requested):
Just gave this a rinse locally and it's pretty solid! 👍. Only ran into one issue, outlined below.

mikesealey commented on packages/client/src/utils/buttonActions.js:
```diff
@@ -527,6 +527,57 @@ const rowActionHandler = async action => {
 }
 
+const delayHandler = async action => {
+  return new Promise(resolve => setTimeout(resolve, action.parameters.duration))
```
That's a great catch. I'll see what I can do.

Maximebaker commented:
Reopen please!!

mikesealey commented:
> Reopen please!!

I've reopened this for now, but it's likely to be easier to start with a fresh PR that achieves the same (and addresses the feedback). 

No timeline available on when this work might take place.
